vendor/bat-native-ads/src/bat/ads/internal/base/url_util_unittest.cc

```diff
@@ -14,6 +14,183 @@
 
 namespace ads {
 
+TEST(BatAdsUrlUtilTest, HttpsScheme) {
```

Comment:
Thanks for adding all of these great test cases.

Here's two more I'd suggest:

- `brave://settings/security` (a valid URL that shouldn't be accepted)
- `http://example.com/brave://wallet`

Reply:
Added, thanks